File: python/S01-elemental-actions/E06-line-intersection.py
def main():
	print("\n\033[35m[========= LINE INTERSECTION =========]\033[0m\n")

	m1: float = float(input("Enter the slope of the first line: "))
	b1: float = float(input("Enter the y-intercept of the first line: "))

	m2: float = float(input("Enter the slope of the second line: "))
	b2: float = float(input("Enter the y-intercept of the second line: "))

	xIntersection: float = (b2 - b1) / (m1 - m2)
	yIntersection: float = m1 * xIntersection + b1

	print(f"\nThe intersection point is at ({xIntersection}, {yIntersection}).")


if __name__ == "__main__":
	main()

File: c++/S01-elemental-actions/E06-line-intersection.cpp
#include <iostream>

int main(int argc, char *argv[]) {
	std::cout << "\n\e[0;35m[========= LINE INTERSECTION =========]\e[0m\n\n";

	float m1, b1, m2, b2, xIntersection, yIntersection;

	std::cout << "Enter values for the first line (m1x+b1): ";
	std::cin >> m1 >> b1;
	std::cout << "Enter values for the second line (m2x+b2): ";
	std::cin >> m2 >> b2;

	xIntersection = (b2 - b1) / (m1 - m2);
	yIntersection = m1 * xIntersection + b1;

	std::cout << "The intersection is: (" << xIntersection << "," << yIntersection << ")" << '\n';

	return 0;
}
